src/p11_child/p11_child_openssl.c

```diff
@@ -1991,7 +1991,26 @@ errno_t do_card(TALLOC_CTX *mem_ctx, struct p11_ctx *p11_ctx,
     /* login: do we need to check for Login Required? */
     if (mode == OP_AUTH) {
         DEBUG(SSSDBG_TRACE_ALL, "Login required.\n");
-        if (pin != NULL) {
+
+        /* Check for protected authentication path */
+        DEBUG(SSSDBG_TRACE_ALL, "Token flags [%lu].\n", token_info.flags);
+        if (token_info.flags & CKF_PROTECTED_AUTHENTICATION_PATH) {
+            DEBUG(SSSDBG_TRACE_ALL, "Protected authentication path.\n");
+
+            rv = module->C_Login(session, CKU_USER, NULL,
```

Comment:
Would it be possible to just set pin=NULL/len=0 instead of copy&pasting entire block?

Reply:
I tried that, but it didn't work for me. 
Also the specification [PKCS11](https://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html) says "To log into a token with a protected authentication path, the pPin parameter to C_Login should be NULL_PTR."

Reply:
> > I tried that, but it didn't work for me.
> 
> What exactly did you try?

I tried setting pin = NULL and calling C_Login similar to your example.
I could not get it to work, hence the copy and paste.

Reply:
I mean, what could be a difference
```
module->C_Login(session, CKU_USER, NULL,
```
vs
```
pin = NULL
module->C_Login(session, CKU_USER, pin,
```
?

Reply:
I agree, it should work, but I failed.
Feel free to change the code to your liking.

Reply:
I have tried it and it works for me. Great!

Reply:
Ok, then please either cherry-pick my additional patch or let's continue in #7064

Reply:
Let's continue in #7064